int32 direction
---
bool response
